implementations/rust/ockam/ockam_vault_aws/src/vault/aws_security_module.rs

```diff
@@ -52,13 +100,105 @@ impl SecurityModule for AwsSecurityModule {
         self.client.sign(key_id, message).await
     }
 
+    /// Verify the signature of a message locally
+    /// This should return the same result as self.client.verify
+    /// The main differences are:
+    ///   - a call to self.client.verify takes more time
+    ///   - a call to self.client.verify can be logged on AWS and benefit from additional access control checks
     async fn verify(
         &self,
         public_key: &PublicKey,
         message: &[u8],
         signature: &Signature,
     ) -> Result<bool> {
-        let key_id = self.get_key_id(public_key).await?;
-        self.client.verify(&key_id, message, signature).await
+        use p256::ecdsa::{signature::Verifier as _, Signature, VerifyingKey};
+
+        let verifying_key =
+            VerifyingKey::from_public_key_der(public_key.data()).map_err(Self::from_pkcs8)?;
+        let ecdsa_signature = Signature::from_der(signature.as_ref()).map_err(Self::from_ecdsa)?;
```

Comment:
I wonder if PublicKeys and Signature have the same format for both Software Vault and AWS Vault. Probably they are if things work, but I haven't checked

Reply:
That is what is tested with [the test below](https://github.com/build-trust/ockam/pull/4978/files/b901a7ee95e0313a7ec56934f905a286ae393c34#diff-755cfc699e456d340eaf60d41309f8be429b740193484b83b8cc0dfd253053a0R178).